scripts/extract-extension.js

```diff
@@ -0,0 +1,52 @@
+const JSZip = require('jszip');
+const { isValidExtensionName } = require('./lib/ExtensionNameValidator');
+const { createWriteStream } = require('fs');
+const { readFile } = require('fs/promises');
+const pipeline = require('util').promisify(require('stream').pipeline);
+
+/**
+ * Extracts exactly one extension into the community extensions folder from a zip file.
+ * @param {string} [zipPath] The path to the zip file to extract.
+ * @param {string} [extensionsFolder] The folder with the extensions.
+ * @returns {Promise<{error: "too-many-files" | "no-json-found"| "invalid-file-name" | "zip-error", details?: any} | {error?: undefined,extensionName: string}>} the name of the extracted extension if successful, else a generic error code.
+ */
+exports.extractExtension = async function (
+  zipPath = '/tmp/ext.zip',
+  extensionsFolder = `${__dirname}/../extensions`
+) {
+  // Load in the archive with JSZip
+  const zip = await JSZip.loadAsync(await readFile(zipPath)).catch((e) => {
+    console.warn(`JSZip loading error caught: `, e);
+    return null;
+  });
+  if (zip === null) return { error: 'zip-error' };
+
+  // Find JSON files in the zip top level folder
+  const jsonFiles = zip.file(/.*\.json$/);
+
+  // Ensure there is exactly 1 file
+  if (jsonFiles.length === 0) return { error: 'no-json-found' };
+  if (jsonFiles.length > 1) return { error: 'too-many-files' };
+  const [file] = jsonFiles;
+
+  if (file.name.slice(-5) !== '.json') return { error: 'invalid-file-name' };
+  const extensionName = file.name.slice(0, -5);
+
+  // Ensure no special characters are in the extension name to prevent relative path
+  // name schenanigans with dots that could put the extension in the reviewed folder
```

Comment:
Ubernit: I think best practices encourage to put a period, comments are sentences:  https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html

```suggestion
  // name schenanigans with dots that could put the extension in the reviewed folder.
```